src/OgAccess.php

```diff
@@ -203,6 +203,10 @@ public static function userAccessEntity($operation, EntityInterface $entity, Acc
       $forbidden = AccessResult::forbidden()->addCacheTags($cache_tags);
       foreach ($groups as $entity_groups) {
         foreach ($entity_groups as $group) {
+          $crud_access = static::userAccessGroupContentEntityCrud($operation, $group, $entity, $user);
+          if ($crud_access->isAllowed()) {
+            return $crud_access->addCacheTags($cache_tags);
+          }
           $user_access = static::userAccess($group, $operation, $user);
           if ($user_access->isAllowed()) {
             return $user_access->addCacheTags($cache_tags);
```

Comment:
I am not 100% sure if we still need this call to `static::userAccess()`. I was not sure of the original use case for this so I haven't touched it. `static::userAccess()` is concerned with groups rather than group content, and the groups have already been dealt with earlier in this method.